model/validate.go

```diff
@@ -66,11 +66,13 @@ func ValidateHost(hostId string, r *http.Request) (*host.Host, int, error) {
 		return nil, http.StatusBadRequest, errors.Errorf("Host %s not found", hostId)
 	}
 	if err != nil {
-		return nil, http.StatusInternalServerError, errors.Wrapf(err, "Error loading context for host %v", hostId)
+		return nil, http.StatusInternalServerError, errors.Wrapf(err, "Error loading context for host %s", hostId)
 	}
-	// if there is a secret, ensure we are using the correct one -- fail if we arent
-	if secret != "" && secret != h.Secret {
-		return nil, http.StatusConflict, errors.Errorf("Invalid host secret for host %v", h.Id)
+	if secret == "" {
+		return nil, http.StatusBadRequest, errors.New("Missing host secret for host %s", h.Id)
```

Comment:
500 seems more appropriate for this one

Reply:
Pushing with 400 after discussion.

Reply:
Actually no, looks like many test failures.